### PR TITLE
feat(status): use helix-fetch instead of request

### LINKED ISSUE #185: request is deprecated; replace with helix-fetch
[request](https://github.com/request/request) is deprecated and should be replaced.
best candidate is [helix-fetch](https://github.com/adobe/helix-fetch)

### PR DESCRIPTION
**Pull request by tripodsan**
fixes #185

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -60,14 +60,86 @@ const getVersion = memoize(async () => (await getPackage()).version || 'n/a');
...
+    response = await context.fetch(uri, options);
+    body = await response.text();
+  } catch (e) {
+    /* istanbul ignore next */
+    if (e instanceof fetchAPI.AbortError) {
+      e.options = options;
+      e.message = 'Error: ETIMEDOUT';
+    }
+    // this is needed, otherwise the sockets hang
+    await context.disconnectAll();
```
Comment (tripodsan) [inline src/index.js:112]: @stefan-guggisberg is this expected ?

File: src/index.js [hunk 2]
```diff
@@ -60,14 +60,86 @@ const getVersion = memoize(async () => (await getPackage()).version || 'n/a');
...
+  const start = Date.now();
+  let response;
+  let body;
+  try {
+    response = await context.fetch(uri, options);
+    body = await response.text();
+  } catch (e) {
+    /* istanbul ignore next */
+    if (e instanceof fetchAPI.AbortError) {
+      e.options = options;
```
Comment (tripodsan) [inline src/index.js:108]: this is a bit ugly, but I didn't want to change the error handling further down...

File: test/fixtures/recordings/Index-Tests_647266286/wrap-function-supports-request-options-check-with-functions_4061816471/recording.har [hunk 3]
```diff
@@ -8,45 +8,58 @@
...
+              "value": "helix-fetch"
+            },
+            {
+              "name": "accept-encoding",
+              "value": "br;q=1, gzip;q=0.8, deflate;q=0.5"
             },
             {
               "name": "referer",
-              "value": "/_status_check/healthcheck.json"
+              "value": "(params) => params.__ow_path"
```
Comment (tripodsan) [inline test/fixtures/recordings/Index-Tests_647266286/wrap-function-supports-request-options-check-with-functions_4061816471/recording.har:36]: I guess, this is wrong.... :-)

File: test/index.test.js [hunk 4]
```diff
@@ -302,12 +329,12 @@ describe('Index Tests', () => {
     const { server } = this.polly;
 
     let ua;
-    server.get('http://localhost/test').intercept((req) => {
+    server.get('http://example.com/test').intercept((req) => {
```
Comment (tripodsan) [inline test/index.test.js:332]: polly + helix-fetch doesn't work on localhost...

File: test/fixtures/recordings/Index-Tests_647266286/wrap-function-supports-request-options-check-with-functions_4061816471/recording.har [hunk 5]
```diff
@@ -4,44 +4,62 @@
...
+            },
+            {
+              "name": "accept-encoding",
+              "value": "br;q=1, gzip;q=0.8, deflate;q=0.5"
             },
             {
-              "name": "content-length",
-              "value": 0
+              "name": "referer",
+              "value": "/_status_check/healthcheck.json"
```
Comment (trieloff) [inline test/fixtures/recordings/Index-Tests_647266286/wrap-function-supports-request-options-check-with-functions_4061816471/recording.har:36]: This is strange.

File: src/index.js (hunk 1 above)
Comment (stefan-guggisberg) [reply src/index.js:100]: > @stefan-guggisberg is this expected ?

no, should not be necessary. I quickly did some ad hoc tests and there wasn't a problem with dangling sockets.

File: test/fixtures/recordings/Index-Tests_647266286/wrap-function-supports-request-options-check-with-functions_4061816471/recording.har (hunk 5 above)
Comment (tripodsan) [reply test/fixtures/recordings/Index-Tests_647266286/wrap-function-supports-request-options-check-with-functions_4061816471/recording.har:36]: why? I changed how the router is mounted.

**Pull request merged**